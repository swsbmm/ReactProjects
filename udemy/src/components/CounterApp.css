*{
    margin: 0;
    padding: 0;
    box-sizing: none;
}

.container {
    background-color: #4b7875;
    width: 100vw;
    height: 100vh;
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
}

p{
    display: flex;
    justify-content: center;
    font-size: 50px;
    font-weight: bold;
}

button{
    padding: 2px;
    background-color: black;
    color: white;
    cursor: pointer;
    margin-right: 5px;
    height: 3rem;
    width: 5rem;
}