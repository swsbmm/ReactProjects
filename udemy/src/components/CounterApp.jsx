import React, { Fragment, useState } from 'react';
import PropTypes from 'prop-types';
// import './CounterApp.css';

export default function CounterApp({ value }) {
  const [count, setCount] = useState(value);
  const sumar = (event) => {
    setCount(count + 1);
  };
  const restar = () => setCount(count - 1);
  const reset = () => setCount(value);

  return (
    <Fragment>
      <div className="container">
        <h1>Counter App</h1>
        <div>
          <p>{count}</p>
          <div>
            <button onClick={sumar}>+1</button>
            <button onClick={restar}>-1</button>
            <button onClick={reset}>reset</button>
          </div>
        </div>
      </div>
    </Fragment>
  );
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
