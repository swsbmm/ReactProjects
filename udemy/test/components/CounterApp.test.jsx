import { render, screen,fireEvent } from "@testing-library/react";
import CounterApp from './../../src/components/CounterApp';

describe('test en el componente <CounterApp/>', ()=> {
    
    const value = 10;

    test('debe hacer mach con el snapshot', ()=> {
        const {container} = render(<CounterApp value={value} />);
        expect(container).toMatchSnapshot();
    })

    test('debe mostrar el valor inicial de 100', () => {
        render(<CounterApp value={value}/>);
        expect(screen.getByText(value)).toBeTruthy();
    })

    test('test con boton: debe incrementar con el boton +1', () => {
        render(<CounterApp value={value}/>);
        fireEvent.click( screen.getByText("+1"))
        expect( screen.getByText("11")).toBeTruthy();
    })

    test('test con boton: debe decrementar con el boton -1', () => {
        render(<CounterApp value={value}/>);
        fireEvent.click( screen.getByText("-1"))
        expect( screen.getByText("9")).toBeTruthy();
    })

    test('test con boton: debe volver al valor:' + value, () => {
        render(<CounterApp value={value}/>);
        fireEvent.click( screen.getByText("+1"))
        fireEvent.click( screen.getByText("+1"))
        fireEvent.click( screen.getByText("+1"))
        fireEvent.click( screen.getByText("reset"))
        expect( screen.getByText("10")).toBeTruthy();
    })
})