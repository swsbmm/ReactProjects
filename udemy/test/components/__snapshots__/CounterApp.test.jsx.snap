// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test en el componente <CounterApp/> debe hacer mach con el snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <h1>
      Counter App
    </h1>
    <div>
      <p>
        10
      </p>
      <div>
        <button>
          +1
        </button>
        <button>
          -1
        </button>
        <button>
          reset
        </button>
      </div>
    </div>
  </div>
</div>
`;
